source/material_model/steinberger.cc

```diff
@@ -383,6 +409,56 @@ namespace aspect
     }
 
 
+    template <int dim>
+    std::vector<double>
+    Steinberger<dim>::
+    phase_volume_fractions (const double      temperature,
+                            const double      pressure,
+                            const std::vector<double> &compositional_fields,
+                            const Point<dim> &) const
+    {
+      std::vector<double> volume_fractions(unique_phase_names.size(), 0.);
+
+      if (material_lookup.size() == 1)
+        {
+          for (unsigned int j = 0; j < unique_phase_names.size(); j++)
+            {
+              if (unique_phase_indices[0][j] != -1)
+                volume_fractions[j] += material_lookup[0]->phase_volume_fraction(unique_phase_indices[0][j],temperature,pressure);
+            }
+
+        }
+      else if (material_lookup.size() == compositional_fields.size() + 1)
```

Comment:
Have you tested that this works with more than one compositional field, and different phase names for the different compositions, with just some overlap?

Reply:
Now I have :)